WalletWasabi/Blockchain/TransactionOutputs/ISmartCoin.cs

```diff
@@ -10,6 +11,8 @@ public interface ISmartCoin
 	/// <remarks>Note there might be a new script type that even NBitcoin does not support.</remarks>
 	ScriptType ScriptType { get; }
 
+	HdPubKey HdPubKey { get; }
```

Comment:
This interface is not ours.

Reply:
How can I access HdPubKey in SelectCoinsForRounds without changing the interface?

Reply:
I don't know, but that's not the only way to go about it? You can also change to something others can supply like address (or pubkey?) Anyway what I meant here is to consult with @Kukks and @onvej-sl about any changes you are intending to do here.

Reply:
In #9252, I introduced a separate coin selector that a wallet can specify, which gave me back access to coins  (amongst other things)

Reply:
Personally, I would add a getter prop to the interface for `Outpoint` instead of the bulky hdpubkey

Reply:
Please introduce address or script pubkey instead of `HdPubKey` in the interface. And can probably remove `ScriptType` from the interface since it can be derived from the address or the script pubkey.

Reply:
Keep `ScriptType` for now and replace HdPubKey with `ScriptPubKey`. @lontivero on the implementation side is that good to use `GetAssumedScriptPubKey`?